### PR TITLE
Implement squash command

### PR DESCRIPTION
**Pull request by kellda**
Allows asking bors to squash all commits before merging.

### CODE REVIEW

File: homu/main.py [hunk 1]
```diff
@@ -1779,6 +1818,10 @@ def main():
         db_query(db, 'SELECT treeclosed_src FROM repos LIMIT 0')
     except sqlite3.OperationalError:
         db_query(db, 'ALTER TABLE repos ADD COLUMN treeclosed_src TEXT')
+    try:
+        db_query(db, 'SELECT squash FROM pull LIMIT 0')
+    except sqlite3.OperationalError:
+        db_query(db, 'ALTER TABLE pull ADD COLUMN squash TEXT')
```
Comment (pietroalbini) [inline homu/main.py:1824]: Is there a reason why this is stored as text instead of integer?

**Review by pietroalbini — CHANGES_REQUESTED** [review]
Hmm, I tried running this locally on my test repo and this exception happened at startup:

```
Traceback (most recent call last):                                                     
  File "/home/pietro/r/github/rust-lang/homu/homu/server.py", line 915, in sync_repo   
    g.my_username, g.repo_labels)                                                      
  File "/home/pietro/r/github/rust-lang/homu/homu/main.py", line 1646, in synchronize  
    command_src=comment.to_json()['html_url'],                                         
  File "/home/pietro/r/github/rust-lang/homu/homu/main.py", line 798, in parse_commands
    state.save()                                                                       
  File "/home/pietro/r/github/rust-lang/homu/homu/main.py", line 366, in save          
    self.squash,                                                                       
AttributeError: 'PullReqState' object has no attribute 'squash'                        
```

File: homu/main.py (hunk 1 above)
Comment (kellda) [reply homu/main.py:1824]: No, I wasn't careful enough when copy-pasting from the `treeclosed_src` migration. I used an `INT` in the `CREATE TABLE`.

**Comment by kellda** [comment]
Should be fixed now

**Review by pietroalbini — APPROVED** [review]
Tested this locally and the PR works great! Thanks for working on it :)

**Comment by Mark-Simulacrum** [comment]
Can we have a follow up PR expanding the docs (not sure where in repo, available at https://bors.rust-lang.org/ though) to include this?

**Comment by pietroalbini** [comment]
I think the docs are in the html templates of this repo.

**Comment by kellda** [comment]
I sent a PR for the docs (#126)

**Pull request merged**